modules/xmpp/ChatRoom.js

```diff
@@ -1026,39 +1005,35 @@ export default class ChatRoom extends Listenable {
      * @param from
      */
     onMessage(msg, from) {
-        const nick
-            = $(msg).find('>nick[xmlns="http://jabber.org/protocol/nick"]')
-                .text()
+        const nick = $_(msg, '>nick[xmlns="http://jabber.org/protocol/nick"]')?.textContent
             || Strophe.getResourceFromJid(from);
 
         const type = msg.getAttribute('type');
 
         if (type === 'error') {
-            const errorMsg = $(msg).find('>error>text').text();
+            const errorMsg = $_(msg, '>error>text').textContent;
 
             this.eventEmitter.emit(XMPPEvents.CHAT_ERROR_RECEIVED, errorMsg);
 
             return true;
         }
 
-        const txt = $(msg).find('>body').text();
-        const subject = $(msg).find('>subject');
+        const txt = $_(msg, '>body')?.textContent;
+        const subject = $_(msg, '>subject');
 
-        if (subject.length) {
-            const subjectText = subject.text();
+        if (subject) {
+            const subjectText = subject.textContent;
 
-            if (subjectText || subjectText === '') {
```

Comment:
Why is the check removed? Any chance the subject has no text content?

Reply:
Because the check doesnt check anything. It is true if subjectText is a string WITH content OR WITHOUT content. Its always a string because `text()` returns `""` for no element as well.